packages/app-mobile/components/FolderPicker.tsx

```diff
@@ -0,0 +1,90 @@
+const React = require('react');
+
+import { FunctionComponent } from 'react';
+import { _ } from '@joplin/lib/locale';
+import Setting from '@joplin/lib/models/Setting';
+import Folder, { FolderEntityWithChildren } from '@joplin/lib/models/Folder';
+const { themeStyle } = require('./global-style.js');
+import Dropdown, { DropdownListItem, OnValueChangedListener } from './Dropdown';
+import { FolderEntity } from '@joplin/lib/services/database/types';
+
+interface FolderPickerProps {
+  disabled?: boolean;
+  selectedFolderId?: string;
+  onValueChange?: OnValueChangedListener;
+  mustSelect?: boolean;
+	folders: FolderEntity[];
+  placeholder?: string;
+  dark?: boolean;
+}
+
+
+const FolderPicker: FunctionComponent<FolderPickerProps> = ({
+	disabled,
+	selectedFolderId,
+	onValueChange,
+	mustSelect,
+	folders,
+	placeholder,
+	dark,
+}) => {
+	const themeId = Setting.value('theme');
+	const theme = themeStyle(themeId);
+
+	const addFolderChildren = (
+		folders: FolderEntityWithChildren[], pickerItems: DropdownListItem[], indent: number
+	) => {
+		folders.sort((a, b) => {
+			const aTitle = a && a.title ? a.title : '';
+			const bTitle = b && b.title ? b.title : '';
+			return aTitle.toLowerCase() < bTitle.toLowerCase() ? -1 : +1;
+		});
+
+		for (let i = 0; i < folders.length; i++) {
+			const f = folders[i];
+			const icon = Folder.unserializeIcon(f.icon);
+			const iconString = icon ? `${icon.emoji} ` : '';
+			pickerItems.push({ label: `${'      '.repeat(indent)} ${iconString + Folder.displayTitle(f)}`, value: f.id });
```

Comment:
Something to test/think about: How well does this work with screen readers? Is it clear while using TalkBack/VoiceOver that this item is a subfolder of the item on the line above it? (I'm not sure whether/how screen readers announce indentation).

Reply:
> I did copy this part pretty much verbatim from components/screens/folder.js

In that case, fixing it everywhere in a future PR would make sense.

> Generally does the app try to be approachable for people using screen readers?

Joplin currently has several bugs that make it very difficult to use with a screen reader. For example, with VoiceOver on iOS, items in the sidebar are selectable regardless of whether the sidebar is closed, making it difficult to navigate to/between notes.

Ideally, Joplin would be accessible, though!